core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -667,15 +667,22 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, Reservation[R, E, A]])
   /**
    * Modifies this `ZManaged` to provide a canceler that can be used to eagerly
    * execute the finalizer of this `ZManaged`. The canceler will run
-   * uninterruptibly, and if completed will cause the regular finalizer to not
-   * run.
+   * uninterruptibly with an exit value indicating that the effect was
+   * interrupted, and if completed will cause the regular finalizer to not run.
    */
   final def withEarlyRelease: ZManaged[R, E, (URIO[R, _], A)] =
+    withEarlyReleaseExit(Exit.interrupt)
+
+  /**
+   * A more powerful version of `withEarlyRelease` that allows specifying an 
+   * exit value in the event of early release.
+   */
+  final def withEarlyReleaseExit(exit: Exit[_, _]): ZManaged[R, E, (URIO[R, _], A)] =
```

Comment:
How about renaming both to `earlyRelease` and `earlyReleaseWith`, this seems to be more in line with other combinators that have a more powerful version.

Reply:
I think maybe `withEarlyRelease` indicates more clearly that it is giving you an option to do early release but not actually doing it? And we do have other `Exit` variants like `finalizer` and `finalizerExit`. But that is super subjective. I'm fine with either one. @iravid What do you think?

Reply:
`withEarlyRelease` is probably more accurate as it "adds" the option to release early